compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -352,7 +352,11 @@ import transform.SymUtils._
             val siteName = site match
               case site: NamedType => site.name.show
               case site => i"$site"
-            s" - did you mean $siteName.$n?$enumClause"
+            val showName =
+              // Add .type to the name if it is a module
+              if n.isClass && n.is(Module) then s"${n.name.show}.type"
```

Comment:
I believe you should get the same result with `.is(ModuleClass)` instead of checking two conditions